com.unity.shadergraph/Editor/Drawing/SearchWindowProvider.cs

```diff
@@ -338,10 +360,19 @@ public bool OnSearcherSelectEntry(SearcherItem entry, Vector2 screenMousePositio
                 if (!(target is ContextView contextView))
                     return true;
 
+                // ensure custom blocks have a unique name provided the existing context.
+                if (blockNode.isCustomBlock)
+                {
+                    HashSet<string> usedNames = new HashSet<string>();
+                    foreach (var other in contextView.contextData.blocks) usedNames.Add(other.value.descriptor.displayName);
+                    blockNode.customName = GraphUtil.SanitizeName(usedNames, "{0}_{1}", blockNode.descriptor.displayName);
+                }
                 // Test against all current BlockNodes in the Context
                 // Never allow duplicate BlockNodes
-                if (contextView.contextData.blocks.Where(x => x.value.name == blockNode.name).FirstOrDefault().value != null)
+                else if (blockNode.isCustomBlock == false && contextView.contextData.blocks.Where(x => x.value.name == blockNode.name).FirstOrDefault().value != null)
```

Comment:
I think this `blockNode.isCustomBlock == false` can be removed as it is already else statement of if (blockNode.isCustomBlock)

Reply:
I think I had left it in for explicitness, but yea- it's redundant.